File: 21:08:2019/Answer.py
class Solution: 
    def longestPalindrome(self, s):
        result = 1
        start = 0
        table = [[False for j in range(len(s))] for i in range(len(s))]
        
        for i in range(len(s)):
            table[i][i] = True
        
        for i in range(len(s)-1):
            if s[i] == s[i+1]:
                table[i][i+1] = True
                start = i
                result = 2

        for k in range(3, len(s)):
            for i in range(len(s) - k + 1):
                j = i + k - 1
                if table[i+1][j-1] and s[i] == s[j]:
                    table[i][j] = True
                    if k > result:
                        result = k
                        start = i

        return (start, result)


print ("Please enter the string: ")
inputString = input()

(start, result) = Solution().longestPalindrome(inputString)

print ("The longest palindrome is: \"{0}\"".format(inputString[ start : start+result ]))

File: 21:08:2019/Answer.java
import java.lang.*;
import java.io.*;
import java.util.*;

class Result {
    private int result, start;
    
    public Result(int result, int start) {
        this.result = result;
        this.start = start;
    }

    public int getResult() {
        return this.result;
    }

    public int getStart() {
        return this.start;
    }
}

public class Answer {
    public static Result longestPalindrome(String s) {
        int result = 1, start = 0;
        boolean table[][] = new boolean[s.length()][s.length()];
        for (int i=0; i<s.length(); i++) {
            table[i][i] = true;
        }

        for (int i=0; i<s.length()-1; i++) {
            if (s.charAt(i) == s.charAt(i+1)) {
                start = i;
                result = 2;
                table[i][i+1] = true;
            }
        }

        for (int k=3; k<s.length(); k++) {
            for (int i=0; i<(s.length() - k + 1); i++) {
                int j = i + k - 1;

                if (table[i+1][j-1] && s.charAt(i) == s.charAt(j)) {
                    table[i][j] = true;
                    if (k > result) {
                        start = i;
                        result = k;
                    }
                }
            }
        }

        return new Result(result, start);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String inputString;

        System.out.print("Please enter the string: ");
        inputString = sc.nextLine();

        Result res = longestPalindrome(inputString);
        System.out.println("The longest palindrome is: \"" + (inputString.substring(res.getStart(), res.getResult() + res.getStart())) + "\"");
        sc.close();
    }
}